app/main.py

```diff
@@ -1,3 +1,19 @@
-def cache(func):
-    # Write your code here
-    pass
+from typing import Callable
+from functools import wraps
+
+
+def cache(func: Callable) -> Callable:
+    storage = {}  # this will be stored in closure because it is nonlocal
+
+    @wraps(func)
+    def wrapper(*args, **kwargs) -> list:
```

Comment:
You can use only rags in this exact task

Reply:
rag - what is it "rag", what do you mean?